src/conv/k-g/k_parser.h

```diff
@@ -25,12 +25,14 @@
 
 #ifndef KPARSER_INCLUDED
 #define KPARSER_INCLUDED
+#define FIRST_NODE 2

```

Comment:
? Where is it used?

Reply:
You see that you are getting into trouble with the primitives names. How about not having a `setName()` but `setBaseName()` and not `getName()` but `getNames()` which returns a list of names? In case of the Bot, it would be the base name with a ".bot" attached. In case of the Arbs, it could be a list of base-name + "." + number + ".arb".

Reply:
> ? Where is it used?

it's not used any more, I was trying to make a loop and I defined it. I will rimove it

Reply:
> You see that you are getting into trouble with the primitives names. How about not having a `setName()` but `setBaseName()` and not `getName()` but `getNames()` which returns a list of names? In case of the Bot, it would be the base name with a ".bot" attached. In case of the Arbs, it could be a list of base-name + "." + number + ".arb".

okay, this seems to be more coherent, I am working on it.

Reply:
the only problem with setBaseName() is in the case of arbs the elements numbers doesn't have to be consequent starting from (1 to n). so if we want to preserve the number of Hex element we have to keep naming the arbs following the .k file. I will update the pull request and then you can tell me what you think

Reply:
I updated with getNames() and setBaseName(), I will change also Arbs::write() and Bot::write().

Reply:
> > ? Where is it used?
> 
> it's not used any more, I was trying to make a loop and I defined it. I will remove it. Edit: it is used in k_parser.cpp to fill the nodes in the vector using a loop
> 
> ![image](https://private-user-images.githubusercontent.com/24756685/346656479-2b0372b8-0bf1-4acc-870a-386f7047edec.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjA5NjUxNDMsIm5iZiI6MTcyMDk2NDg0MywicGF0aCI6Ii8yNDc1NjY4NS8zNDY2NTY0NzktMmIwMzcyYjgtMGJmMS00YWNjLTg3MGEtMzg2ZjcwNDdlZGVjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzE0VDEzNDcyM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWIzOGYxYzY2YzBlMjVjMmYyYTczM2VmYjU4MzYwOGE1YTU0YmY0YTYxM2VhMTAzNTA5MGE0YjQwYzQzZDhhMDImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.DfIyU04S5DWUA4B9WxB9p7lcppziHWMijdjqClnAEtM)
> 
> I can just use i_n+2 because it's a fixed format, but this seamed nicer. what do you think?

Okay, that's a reason. However, in this case, it should be kept in k_parser.cpp (declaration/definition) and a `const size_t`, e.g.
```c++
const size_t FirstNode = 2;
```